src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/CacheEntry.java

```diff
@@ -15,35 +15,588 @@
+            if (cacheControl == null)
+            {
+                return false;
+            }
+            final CacheControl parsedCacheControl = cache.responseCacheControlFW.parse(cacheControl);
+            return parsedCacheControl.contains("private");
+        }
+    }
+
+    public boolean isUpdateRequestForThisEntry(ListFW<HttpHeaderFW> requestHeaders)
+    {
+        return CacheUtils.isMatchByEtag(requestHeaders, this.cachedRequest.etag());
+    }
+
+
+    public boolean subscribeToUpdate(OnUpdateRequest onModificationRequest)
+    {
+        final boolean polling = this.state == REFRESHING || this.state == CAN_REFRESH;
+        if (polling)
+        {
+            this.subscribers.add(onModificationRequest);
+        }
+        return polling;
+    }
 
-    void serveClient(AnswerableByCacheRequest streamCorrelation);
 
-    void cleanUp();
+    public void subscribers(Consumer<OnUpdateRequest> consumer)
+    {
+        subscribers.stream().forEach(consumer);
+        subscribers.clear();
+    }
 
-    boolean canServeRequest(ListFW<HttpHeaderFW> request, short authScope);
 
-    boolean isUpdateRequestForThisEntry(ListFW<HttpHeaderFW> requestHeaders);
+    public boolean isUpdateBy(CacheableRequest request)
+    {
+        ListFW<HttpHeaderFW> responseHeadersRO = request.getResponseHeaders(cache.responseHeadersRO);
+        MutableDirectBuffer cachedResponsePayload = getCachedData();
+        MutableDirectBuffer responsePayload = request.getData(cache.responseBufferPool);
+        int cachedHeaderSize = cachedRequest.responseHeadersSize();
+        int headerSize = request.responseHeadersSize();
+        int cachedResponseSize = cachedRequest.responseSize();
+        int responseSize = request.responseSize();
 
-    boolean subscribeToUpdate(OnUpdateRequest onModificationRequest);
+        int cachedPayloadSize = cachedResponseSize - cachedHeaderSize;
+        int payloadSize = responseSize - headerSize;
 
-    Stream<OnUpdateRequest> subscribersOnUpdate();
+        String status = HttpHeadersUtil.getHeader(responseHeadersRO, HttpHeaders.STATUS);
+        if (status.startsWith("304"))
+        {
+            return false;
+        }
+        if (cachedPayloadSize == payloadSize)
+        {
+            for (int i = 0, length = payloadSize; i < length; i++)
+            {
+                if (cachedResponsePayload.getByte(cachedHeaderSize + i) != responsePayload.getByte(headerSize + i))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+        return true;
+    }
 
-    boolean isUpdateBy(CacheableRequest request);
+    private MutableDirectBuffer getCachedData()
+    {
+        return this.cachedRequest.getData(cache.cachedResponseBufferPool);
+    }
 
-    void refresh(AnswerableByCacheRequest request);
+    public void refresh(AnswerableByCacheRequest request)
+    {
+        if (request == pollingRequest)
```

Comment:
Please confirm that this intended to be an identity check rather than an `equals` check.

Reply:
confirmed